core/src/main/java/com/opensymphony/xwork2/ognl/OgnlValueStack.java

```diff
@@ -246,7 +247,7 @@ public Object findValue(String expr, boolean throwExceptionOnFailure) {
     }
 
     protected void setupExceptionOnFailure(boolean throwExceptionOnFailure) {
-        if (throwExceptionOnFailure) {
+        if (throwExceptionOnFailure || (devMode && logMissingProperties)) {
```

Comment:
I think this is the main reason why `devMode` and `logMissingProperties` should continue to be "linked" (at least in 2.5.x).  Most users probably only want to check for missing properties in devMode (development) anyway.

If `logMissingProperties` becomes a "standalone flag", it could turn on `THROW_EXCEPTION_ON_FAILURE` in a production configuration - and that could have greater repercussions than intended.  Maybe it could be decoupled in 2.6 in a "safe way" ?

Reply:
`throwExceptionOnFailure || logMissingProperties` also works even better. As far as I can see in codes, Struts will check them separately later to check throw any exception and log any message.